test/minitest/test_minitest_mock.rb

```diff
@@ -495,4 +495,18 @@ def test_mock_with_yield
     @tc.assert_equal true, rs
   end
 
+  def test_blow_up_on_multiple_stub
+    h = {}
+    e = assert_raises StubError do
+      h.stub(:empty?, false) do
+        h.stub(:empty?, true) do
```

Comment:
@zenspider The real scenario was with arguments which makes a bit more sense. I agree that here it looks like it is just reversing the previous stub. 

Maybe my test is not good enough because it does not show a believable scenario. Do you think I should change the test and use arguments?

Now possible scenarios are:

RSpec doesn't do it with blocs (afaik) but stubs are a one off, one call. So if you expect 2 calls with different arguments, in RSpec you have to stubs. They also have to happen in the same order. So the most mindless translation would be the nested stubs in Minitest. Let say you are building a converter, this is not such a bad exception to raise because you see immediately what is wrong.

But if you go further, one can create a method which hides the stubbing and yield in her/his tests. In which case the stubs are not on top of each other and can be missed.